### PR TITLE
Add TFROBOT to docs

### PR DESCRIPTION
**Pull request by Mik-TF**
# Related Issue

To answer issue #372 

# Work Done

- Documentation based on the tfgrid-sdk-go mass-deployer docs

### CODE REVIEW

**Comment by Mik-TF** [comment]
Updated as per the current tfrobot repo.

File: src/tfrobot/tfrobot_commands_flags.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+You can run the command `tfrobot help` at any time to access the help section. This will also display the available commands.
+
+| Command    | Description                                                |
+| ---------- | ---------------------------------------------------------- |
+| completion     | Generate the autocompletion script for the specified shell                        |
+| help       | Help about any command                                     |
+| version    | Get latest build tag                                       |
+
+You can also use the following flags: `-c` (`--config`) and `-h` (`--help`).
```
Comment (A-Harby) [inline src/tfrobot/tfrobot_commands_flags.md:27]: This part feels repeated; you already mentioned them down on the flag table.

File: src/tfrobot/tfrobot_commands_flags.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+You can use different flags to configure your deployment.
+
+| Flag | Usage |
+| :---:   | :---: |
+| -c | used to specify path to configuration file |
+| -o | used to specify path to output file to store the output info in |
+| -d | allow debug logs to appear in the output logs |
+| -h | help |
+
+> **Note:** Make sure to use every flag once. If the flag is repeated, it will ignore all values and take the last value of the flag.
```
Comment (A-Harby) [inline src/tfrobot/tfrobot_commands_flags.md:59]: Can you also add https://github.com/threefoldtech/tfgrid-sdk-go/blob/development/tfrobot/README.md#logs

File: src/tfrobot/tfrobot_config.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+## Introduction
+
+To use TFROBOT, the user needs to create a YAML or a JSON configuration file that will contain the mass deployment information, such as the groups information, number of VMs to deploy how, the compute, storage and network resources needed, as well as the user's credentials, such as the SSH public key, the network (main, test, dev, qa) and the TFChain mnemonics.
+
+## Examples
+
+We present here a configuration file example that deploys a 3 nodes, with 2 vcores, 16GB of RAM, 100GB of SSD and 50GB of HDD. The same deployment is shown with a YAML file and with a JSON file. Parsing is based on file extension, TFROBOT will use JSON format if the file has a JSON extension and YAML format otherwise.
+
+You can use this example for guidance, and make sure to replace placeholders and adapt the groups based on your actual project details. To the minimum, `ssh_key1` should be replaced by the user SSH public key and `example-mnemonic` should be replaced by the user mnemonics.
+
```
Comment (A-Harby) [inline src/tfrobot/tfrobot_config.md:22]: Can you also add any kind of IPs in the example (ygg, ipv4, or ipv6)?

File: src/tfrobot/tfrobot_configurations.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+| env_vars | map of env vars | map of type string to string |
+| ssd | list of disks | should be of type disk|
+| root_size | root size in GB | 0 for default root size, max 10TB |
+
+## Disk
+
+| Field | Description| Supported Values|
+| :---:   | :---: | :---: |
+| Size | disk size in GB| positive integer min = 15 |
+| Mount | disk mount point | path to mountpoint |
```
Comment (A-Harby) [inline src/tfrobot/tfrobot_configurations.md:68]: its mount_point not Mount, I already open an issue for it to be updated in the docs.

Comment (Mik-TF) [reply src/tfrobot/tfrobot_configurations.md]: Done. changed Mount to mount_point and Size to size (so it's all lower caps)

File: src/tfrobot/tfrobot_config.md (hunk 3 above)
Comment (Mik-TF) [reply src/tfrobot/tfrobot_config.md:22]: Done. Added ipv4 true, ipv6 false.

File: src/tfrobot/tfrobot_commands_flags.md (hunk 2 above)
Comment (Mik-TF) [reply src/tfrobot/tfrobot_commands_flags.md:58]: I added logs in this page but note that it was already on another page (Deployment page)

File: src/tfrobot/tfrobot_commands_flags.md (hunk 1 above)
Comment (Mik-TF) [reply src/tfrobot/tfrobot_commands_flags.md]: Done. If I understood correctly, I removed the line `You can also use the following flags: `-c` (`--config`) and `-h` (`--help`).`

**Comment by Mik-TF** [comment]
Took the opportunity to fix a broken link in grid_stacks.

File: src/tfrobot/tfrobot_commands_flags.md (hunk 2 above)
Comment (A-Harby) [reply src/tfrobot/tfrobot_commands_flags.md:58]: Oh, I saw it. I agree, it would be enough on one page.

Comment (Mik-TF) [reply src/tfrobot/tfrobot_commands_flags.md]: Alright. Just removed it from commands_flags. Only one instance now.

File: src/tfrobot/tfrobot_config.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+```
+node_groups:
+  - name: group_a
+    nodes_count: 3
+    free_cpu: 2
+    free_mru: 16
+    free_ssd: 100
+    free_hdd: 50
+    pubip4: true
+    pubip6: false
```
Comment (A-Harby) [inline src/tfrobot/tfrobot_config.md:34]: - You have added the public IPs in the node group (which works as a node filter), but in the VMs group you didn't add them, so you will filter for nodes with public IPs only to deploy a VM with no IPs.
- And can you also add a note if it isn't here that if no IPs are specified as true, the bot will automatically add ygg ip to the VM (at least one IP should be true to be able to ssh to the VM).
- Its `public_ip4` not pubip4, and `public_ip6` not pubip6.

**Review by A-Harby — CHANGES_REQUESTED** [review]

File: src/tfrobot/tfrobot_config.md (hunk 5 above)
Comment (Mik-TF) [reply src/tfrobot/tfrobot_config.md]: Done. Took the opportunity to add dedicated, certified and region. Let me know if things are off!

Thanks.

**Review by xmonader — APPROVED** [review]

**Pull request merged**